src/plugins/intel_gpu/src/kernel_selector/cl_kernels/fully_connected_gpu_bf_tiled.cl

```diff
@@ -218,18 +246,38 @@ KERNEL(fc)(
                     }
                 }
             #endif
-            weights_offset += TILE_K_OFM * SIMD;
+            weights_offset += TILE_K_OFM_PACKED * SIMD;
 
             unroll_for (uint kii = 0; kii < TILE_K; ++kii) {
                 const uint total_k = ki * TILE_K + kii;
                 unroll_for (uint bi = 0; bi < TILE_B; ++bi) {
                     INPUT0_TYPE in_val = _sub_group_shuffle(((INPUT0_TYPE*)(&in_0[bi]))[total_k / SIMD], total_k % SIMD);
                     unroll_for (uint fi = 0; fi < TILE_OFM; ++fi) {
+#if DECOMPRESSION_SCALE_POST_OP
+                        ((ACCUMULATOR_TYPE*)(&acc_tmp[bi]))[fi] += in_val * ((ACCUMULATOR_TYPE*)(&wei))[kii * TILE_OFM + fi];
+#else
                         ((ACCUMULATOR_TYPE*)(&acc[bi]))[fi] += in_val * ((ACCUMULATOR_TYPE*)(&wei))[kii * TILE_OFM + fi];
+#endif
                     }
                 }
             }
         }
+#if DECOMPRESSION_SCALE_POST_OP
+        unroll_for (uint bi = 0; bi < TILE_B; ++bi) {
+            unroll_for(uint fi = 0; fi < TILE_OFM; ++fi) {
+                const uint offset_ofm = out_f + fi*SIMD + sglid;
+
+                #if DECOMPRESSION_SCALE_GROUPS_NUM > 1
+                    const uint scale_offset = (offset_ofm % DECOMPRESSION_SCALE_BATCH_NUM) * DECOMPRESSION_SCALE_BATCH_PITCH +
```

Comment:
offset_ofm is required only here

Reply:
Not really. It may be used for both scales and zero points offset compute. Anyway, if it's not used by the code below due to shape configuration, it will be removed to the compiler as dead code and will not impact the performance.